x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseUtils.java

```diff
@@ -63,27 +63,23 @@ public static boolean licenseNeedsExtended(License license) {
      * Checks if the signature of a self generated license with older version needs to be
      * recreated with the new key
      */
-    public static boolean signatureNeedsUpdate(License license, DiscoveryNodes currentNodes) {
+    public static boolean signatureNeedsUpdate(License license, ClusterState currentState) {
         assert License.VERSION_ENTERPRISE == License.VERSION_CURRENT : "update this method when adding a new version";
 
         String typeName = license.type();
         return (LicenseType.isBasic(typeName) || LicenseType.isTrial(typeName)) &&
         // only upgrade signature when all nodes are ready to deserialize the new signature
             (license.version() < License.VERSION_CRYPTO_ALGORITHMS
-                && compatibleLicenseVersion(currentNodes) >= License.VERSION_CRYPTO_ALGORITHMS);
+                && compatibleLicenseVersion(currentState) >= License.VERSION_CRYPTO_ALGORITHMS);
     }
 
-    public static int compatibleLicenseVersion(DiscoveryNodes currentNodes) {
-        return getMaxLicenseVersion(currentNodes.getMinNodeVersion());
+    public static int compatibleLicenseVersion(ClusterState state) {
+        return getMaxLicenseVersion(state.getMinTransportVersion());
     }
 
-    public static int getMaxLicenseVersion(Version version) {
-        if (version != null && version.before(Version.V_7_6_0)) {
-            return License.VERSION_CRYPTO_ALGORITHMS;
-        } else {
-            assert License.VERSION_ENTERPRISE == License.VERSION_CURRENT : "update this method when adding a new version";
-            return License.VERSION_ENTERPRISE;
-        }
+    public static int getMaxLicenseVersion(TransportVersion compatibleWith) {
```

Comment:
The compatibleWith arg looks unused?

Reply:
I can remove the arg, but I was split on how much of this code to take out overall. A lot of the version compatibility stuff isn't necessary anymore since we haven't made any big changes in the 8.x series, but if we take it out we'd just have to put it back in when we do. Do you have an opinion on how much of that we should rip out? I could expand the blast radius of this PR a bit and clean up some more code that isn't being used.

Reply:
I think version can be removed from here. If we need to conditionalize in the future, we should use things like the feature api.

Reply:
👍🏼 I'll rip out the code we'd need to rewrite anyway to use the feature API at least.